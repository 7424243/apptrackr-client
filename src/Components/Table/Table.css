table {
    margin-left: auto;
    margin-right: auto;
    border-top: 1px solid black;
    border-left: 1px solid black;
    border-right: 1px solid black;
}

th {
    text-align: left;
    border-bottom: 1px solid black;
    padding: 3px;
    border-right: 1px solid black;
}

.last_th {
    border-right: none;
}
