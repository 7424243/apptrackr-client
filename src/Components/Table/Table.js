import React, {Component} from 'react'
import ApptrackrContext from '../../ApptrackrContext'
import TableItem from '../TableItem/TableItem'
import config from '../../config'
import TokenService from '../../services/token-service'
import './Table.css'

class Table extends Component {

    state = {
        applications: [],
        error: null
    }

    static contextType = ApptrackrContext

    componentDidMount() {
        const {user_id} = this.context
        fetch(`${config.API_ENDPOINT}/applications/user/${user_id}`, {
            method: 'GET',
            headers: {
                'content-type': 'application/json',
                'Authorization': `bearer ${TokenService.getAuthToken()}`
            }
        })
            .then(res => {
                if(!res.ok) {
                    return res.json().then(e => Promise.reject(e))
                }
                return res.json()
            })
            .then(applications => {
                this.setState({applications})
                this.context.getUserApplications(applications)
            })
            .catch(error => {
                this.setState({error})
                console.error({error})
            })
    }
    
    render() {
        const {applications} = this.state
        const filter = this.props.filter
        const sortFunction = (a, b) => {
            if(a.job_name.toLowerCase() < b.job_name.toLowerCase()) {
                return -1
            }
            if(a.job_name.toLowerCase() > b.job_name.toLowerCase()) {
                return 1
            }
            return 0
        }
        const alphabetizedApplications = applications.sort(sortFunction)
        const applicationItems = alphabetizedApplications.map(application => {
            return (
                <TableItem
                    key={application.id}
                    id={application.id}
                    job={application.job_name}
                    company={application.company_name}
                    dateApplied={application.date_applied ? application.date_applied : null}
                    interviewDate={application.interview_date ? application.interview_date : null}
                    status={application.status}
                />
            )
        })
        const filteredApplications = alphabetizedApplications.filter(application => application.status === filter)
        const filteredItems = filteredApplications.map(application => {
            return (
                <TableItem
                    key={application.id}
                    id={application.id}
                    job={application.job_name}
                    company={application.company_name}
                    dateApplied={application.date_applied ? application.date_applied : null}
                    interviewDate={application.interview_date ? application.interview_date : null}
                    status={application.status}
                />
            )
        })

        return (
            <table>
                <thead>
                    <tr>
                        <th>Job</th>
                        <th>Company</th>
                        <th>Date Applied</th>
                        <th>Interview Date</th>
                        <th className='last_th'>Status</th>
                    </tr>
                </thead>
                {filter !== 'All' ? filteredItems : applicationItems}
            </table>
        )
    }
}

export default Table