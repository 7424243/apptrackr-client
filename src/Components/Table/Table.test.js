import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom'
import Table from './Table'

describe('Table Component', () => {

    //smoke test
    it('renders without crashing', () => {
        Storage.prototype.getItem = jest.fn(() => 'bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoyLCJpYXQiOjE2MTE5NTYwMDMsInN1YiI6ImRlbW8ifQ.f9WIVijmP-pud7JsTd2ysx4eqtpm6JnzOkUeGRihdVE')
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><Table/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        Storage.prototype.getItem = jest.fn(() => 'bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoyLCJpYXQiOjE2MTE5NTYwMDMsInN1YiI6ImRlbW8ifQ.f9WIVijmP-pud7JsTd2ysx4eqtpm6JnzOkUeGRihdVE')
        const tree = renderer
            .create(<BrowserRouter><Table/></BrowserRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    
})