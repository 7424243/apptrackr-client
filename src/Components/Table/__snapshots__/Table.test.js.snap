// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component renders the UI as expected 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Job
      </th>
      <th>
        Company
      </th>
      <th>
        Date Applied
      </th>
      <th>
        Interview Date
      </th>
      <th
        className="last_th"
      >
        Status
      </th>
    </tr>
  </tbody>
</table>
`;
